docker/secondary-analysis-python/tools/submit/submit/stage.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+
+"""
+Note:
+This is a slightly modified version of a script from the staging service repo here:
+https://github.com/HumanCellAtlas/staging-service/blob/master/scripts/stage_file.py
+
+It has been modified to facilitate calling it from our submission WDL.
+
+This was a short term solution for demo purposes but we don't want to keep two
+copies of this script in the long term. In future, we expect to delete this
+script because one of the following things will happen:
+1) We'll merge our changes into the staging service repo and delete this copy.
+OR
+2) More likely, future changes to the staging service will make this script unnecessary.
+
+
+Stage files in the HCA Staging Area
+
+Usage:
+
+    stage_file cp <file> <urn>
+
+"""
+
+import argparse, base64, json, os, sys
+from collections import deque
+
+try:
+    import boto3
+    from boto3.s3.transfer import TransferConfig
+except ImportError:
+    print("\nPlease install boto3 to use this script, e.g. \"pip install boto3\"\n")
+    exit(1)
+
+KB = 1024
+MB = KB * KB
+
+
+def sizeof_fmt(num, suffix='B'):
+    """
+    From https://stackoverflow.com/a/1094933
+    """
+    for unit in ['', 'Ki', 'Mi', 'Gi', 'Ti', 'Pi', 'Ei', 'Zi']:
+        if abs(num) < 1024.0:
+            return "%d %s%s" % (num, unit, suffix)
+        num /= 1024.0
+    return "%.1f %s%s" % (num, 'Yi', suffix)
+
+
+class Main:
+
+    STAGING_BUCKET_TEMPLATE = "org-humancellatlas-staging-%s"
+    CLEAR_TO_EOL = "\x1b[0K"
+
+    def __init__(self):
+        self._parse_args()
+        self._parse_urn(self.args.urn)
+        session = boto3.session.Session(**self.aws_credentials)
+        self.s3 = session.resource('s3')
+        self._stage_file(self.args.file_path)
+
+    def _parse_args(self):
+        parser = argparse.ArgumentParser(description=__doc__,
+                                         formatter_class=argparse.RawDescriptionHelpFormatter)
+        parser.add_argument('file_path', metavar="<file>",
+                            help="name of file to stage")
+        parser.add_argument('urn', metavar='<URN>',
+                            help="URN of staging area (given to you by Ingest Broker)")
+        self.args = parser.parse_args()
+
+    def _parse_urn(self, urn):
+        # TODO reimplement as deque
+        urnbits = urn.split(':')
+        if len(urnbits) == 6:
```

Comment:
I'm not familiar with the format of an `urn`, or when one would have 5 items vs. 6, so adding a docstring with examples would be helpful here. This can go in as a later improvement to the original script though!